### PR TITLE
PendingReleaseNotes: Add note for POOL_APP_NOT_ENABLED

### PR DESCRIPTION
**Pull request by pdvian**
Adds release notes for the fix added in #47560

Signed-off-by: Prashant D <pdhange@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by ljflores — APPROVED** [review]
Do we need to add any directive on silencing/fixing the warning? Or perhaps that is implied.

File: PendingReleaseNotes [hunk 1]
```diff
@@ -57,6 +57,9 @@
   level storage operator (like Rook) to recreate the missing file system.
   See https://docs.ceph.com/en/latest/cephfs/administration/#file-systems
   docs for more information.
+* RADOS: A POOL_APP_NOT_ENABLED health warning will now be reported if
```
Comment (neha-ojha) [inline PendingReleaseNotes:60]: maybe add a sentence to say what users need to do to resolve or mute the warning?

**Review by neha-ojha — APPROVED** [review]

**Review by ljflores — APPROVED** [review]
LGTM!

**Comment by ljflores** [comment]
jenkins test api

File: PendingReleaseNotes [hunk 2]
```diff
@@ -57,6 +57,12 @@
...
+  the application is not enabled for the pool irrespective of whether
+  the pool is in use or not. Always add ``application`` label to a pool
```
Comment (idryomov) [inline PendingReleaseNotes:62]: I think these are more commonly referred to as application tags, not labels.  It's `tag` in the OSD cap syntax too:
```
  There are two alternative forms of the optional ``{match-spec}`` syntax::

        pool={pool-name} [namespace={namespace-name}] [object_prefix {prefix}]

        [namespace={namespace-name}] tag {application} {key}={value}
```
```suggestion
  the pool is in use or not. Always tag a pool with an application using ``ceph osd pool application enable`` command
```

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
@pdvian I think we need a backport for 18.2.1 and a retroactive backport for 17.2.7 here.

**Pull request merged**